bundles/org.openhab.binding.vesync/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,343 @@
+				<context>text</context>
+				<label>MAC Id</label>
+				<description>The MAC Id of the device as reported by the API.</description>
+			</parameter>
+			<parameter name="deviceName" type="text">
+				<context>text</context>
+				<label>Device Name</label>
+				<description>The name allocated to the device by the app. (Must be unique if used)</description>
+			</parameter>
+		</config-description>
+
+	</thing-type>
+
+	<channel-type id="deviceEnabledType">
+		<item-type>Switch</item-type>
+		<label>Switched On</label>
+		<description>Indicator if the device is switched on</description>
+		<state readOnly="false"/>
+	</channel-type>
+
+	<channel-type id="deviceChildLockEnabledType">
+		<item-type>Switch</item-type>
+		<label>Display Lock</label>
+		<description>Indicator if the devices child lock is enabled (Display Lock)</description>
+		<state readOnly="false"/>
+	</channel-type>
+
+	<channel-type id="deviceDisplayEnabledType">
+		<item-type>Switch</item-type>
+		<label>Display</label>
+		<description>Indicator if the devices display is enabled</description>
+		<state readOnly="false"/>
+	</channel-type>
+
+	<channel-type id="deviceFilterLifePercentageType">
+		<item-type>Number:Dimensionless</item-type>
+		<label>Filter Life Remaining</label>
+		<description>Indicator of the remaining filter life</description>
+		<state readOnly="true" pattern="%.0f %%"/>
+	</channel-type>
+
+	<channel-type id="airPurifierModeType">
+		<item-type>String</item-type>
+		<label>Operation mode</label>
+		<description>The operating mode the air purifier is currently set to</description>
+		<state readOnly="false">
+			<options>
+				<option value="auto">Auto</option>
+				<option value="manual">Manual Fan Control</option>
+				<option value="sleep">Sleeping Auto</option>
+			</options>
+		</state>
+	</channel-type>
+
+	<channel-type id="deviceAFNightLight">
+		<item-type>String</item-type>
+		<label>Night Light</label>
+		<description>The operating mode of the night light functionality</description>
+		<state readOnly="false">
+			<options>
+				<option value="on">On</option>
+				<option value="dim">Dimmed</option>
+				<option value="off">Off</option>
+			</options>
+		</state>
+	</channel-type>
+
+
+	<channel-type id="airPurifierFanLevelType">
+		<item-type>Number:Dimensionless</item-type>
+		<label>Fan Speed</label>
+		<description>Indicator of the current fan speed</description>
+		<state readOnly="true" pattern="%.0f"/>
+	</channel-type>
+
+	<channel-type id="deviceErrorCodeType">
+		<item-type>Number:Dimensionless</item-type>
+		<label>Device Error Code</label>
+		<description>Indicator of the current error code of the device</description>
+		<state readOnly="true" pattern="%.0f"/>
+	</channel-type>
+
+	<channel-type id="deviceAirQualityBasicType">
+		<item-type>Number:Dimensionless</item-type>
+		<label>Air Quality</label>
+		<description>System representation of air quality</description>
+		<state readOnly="true" pattern="%.0f"/>
```

Comment:
Display the QuantityType's unit. Please check all.
```suggestion
		<state readOnly="true" pattern="%.0f %unit%"/>
```

Reply:
I've updated the pattern's to reflect the units, where I know them. Nearly all are dimensionless as I couldn't find the correct mappings, but the pattern's have been updated to show the units as reported by the external system. Hopefully this is ok.

Reply:
Hi @fwolter, My apologies for the delay in jumping back on this - work and family commitments meant I had to handle those for a while. In regards to the unit's, I have just found our the air quality unit is [PM2.5 (Particulate Matter)](https://www.epa.gov/pm-pollution/particulate-matter-pm-basics) (I swear the app text must have changed to PM2.5 from PPM2.5). I have looked through the unit's and cannot see one this would map to. Is there an obvious match I'm missing otherwise it looks like this should possibly be Dimensionless with a pattern of `<state readOnly="true" pattern="%.0f PM2.5"/>`, but I just wanted to check there wasn't some match I had completely missed? Thanks again.

Reply:
Changed to what I think is necessary will await feedback, if I've missed some way of mapping PM2.5 🤞 I have not.

Reply:
Glad you figured out the correct units. Seems like I was wrong, too. The second P was indeed a bit misleading.